iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -548,17 +548,12 @@ protected override void Dispose(bool disposing)
             {
                 base.Dispose(disposing);
 
-                if (_disposed)
-                {
-                    return;
-                }
-
                 Logging.Info(this, $"{nameof(disposing)}");
 
                 if (disposing)
                 {
                     _closed = true;
-                    AmqpUnitManager.GetInstance().RemoveAmqpUnit(_amqpUnit);
+                    AmqpUnitManager.GetInstance()?.RemoveAmqpUnit(_amqpUnit);
                     _disposed = true;
```

Comment:
if we never use the _disposed field, we should just remove it.

Reply:
nvm. Talked to James offline and he explained it to me. 
Based on our conversation, we can still keep the if(_disposed) check if we move it above the base.Dispose() call.